xdsl/parser/core.py

```diff
@@ -792,6 +797,32 @@ def _parse_op_result_list(self) -> list[tuple[Span, int]]:
     def parse_optional_attr_dict(self) -> dict[str, Attribute]:
         return self.parse_optional_dictionary_attr_dict()
 
+    def resolve_operands(
+        self, args: list[UnresolvedOperand], input_types: ArrayAttr[Attribute],
```

Comment:
```suggestion
        self, args: Sequence[UnresolvedOperand], input_types: Sequence[Attribute],
```
That way, this will also work with list, tuples, and ArrayAttr.

Reply:
I tried to make input_types a Sequence, but I currently get the following pyright error:

```
Argument of type "ParameterDef[ArrayAttr[Attribute]]" cannot be assigned to parameter "input_types" of type "Sequence[Attribute]" in function "resolve_operands"
```

Are you sure this is a Sequence type? I guess we could make it one but maybe that would be a separate PR?

Reply:
Oh, indeed I think at some point I might have seen a similar problem.
Can we switch it back to `Sequence` then, and instead pass `inputs.data` when calling it?
That way it should still work with everything else!

Reply:
I used `Iterable` instead.

Reply:
`Iterable` cannot work because it does not have a `__len__` :/